File: 5. Longest Palindromic Substring/Solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        def palindrome(s, l, r):
            while (l >= 0 and r < len(s) and s[l] == s[r]):
                l -= 1
                r += 1
            return s[l+1:r]
        
        max_len = ""
        for i in range(len(s)):
            sub_str1 = palindrome(s, i, i)
            sub_str2 = palindrome(s, i, i + 1)
            if len(max_len) <= len(sub_str1):
                max_len = sub_str1
            if len(max_len) <= len(sub_str2):
                max_len = sub_str2
        return max_len

File: 5. Longest Palindromic Substring/Solution.java
class Solution {
    public String longestPalindrome(String s) {
        String res = "";
        
        for (int i = 0; i < s.length(); i++) {
            String str1 = palindrome(s, i, i);
            String str2 = palindrome(s, i, i + 1);
            String str = str1.length() > str2.length() ? str1 : str2;
            res = res.length() > str.length() ? res : str;
        }
        return res;
    }
    
    public String palindrome(String s, int l, int r) {
        while (l >= 0 && r < s.length() && s.charAt(l) == s.charAt(r)) {
            l--;
            r++;
        }
        return s.substring(l + 1, r);
    }
}